### DISCUSSION TITLE
Setting extra flags for the cluster (namely "--max-pods")

### DISCUSSION THREAD

**Discussion by MTRNord** [kube-hetzner/terraform-hcloud-kube-hetzner#234] [Q&A]
Hi, I plan to migrate my existing cluster in whole to this setup. I currently have 231 pods running, which is more than the default pod limit. To up this one usually adds `--max-pods` to the kubelet. How would I do this with this setup to be able to run all the pods? :)

**Comment by mysticaltech** [comment]
@MTRNord The default seems to be 110 indeed. This stack overflow explains how to do it https://stackoverflow.com/questions/65894616/how-do-you-increase-maximum-pods-per-node-in-k3s. Basically, for this project, you need to create a PR that:

1/ Create a `kubelet` config file on the machine, via the file provisioned and `yamlencode`, as done for other files in https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/init.tf.

2/ Change, the k3s config, in https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/init.tf#L18, https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/control_planes.tf#L60, and https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/agents.tf#L55 to pass them the `kubelet` config path location as a third argument in the form of `config=<KUBELET_CONFIG_FILE_LOCATION>`.

3/ Finally, make sure the above logic is only applied based on a new variable, that ideally would contain the kubeletConfig info. And would support any options, not only maxPods. See https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1/, as to make this as flexible as possible. But that last part is a plus, if you want to just create a variable with with maxPods and just test for it to apply the new config, it would work too.
